### ISSUE TITLE
New Websocket created every second

### ISSUE THREAD

**Issue by bazz1tv** [loopy750/SRT-Stats-Monitor#1]
Dear Loopy,

First, thanks so much for this great open source project!! 🙌 

🧐 during streams, I noticed an odd OBS-Websocket connection being opened and closed every second. I traced it back to SRT Stats Monitor. It seems bad, despite harmless. It is also bloating the OBS log files.

Why not have SRT Monitor open a single web socket for the duration of the program's lifetime? ✨ 

SRT Stats Monitor's current implementation calls separate nodeJS scripts that each open and close their own websocket every single time they are run. I wonder how to make it work for a persistent socket, but it seems the entire approach would need to change.

You could consolidate the nodeJS scripts into a single script that opens the websocket once, and then communicates with the main program on a named pipe to receive the commands to run. 🤔 

Hope this helps
Faithfully,
Bazz

**Comment by loopy750** [comment]
You are correct, it does cause a lot of "connection spam", and I wish it didn't.

It is, however, harmless, and is reliable. Currently the program externally executes node which then runs the js file. While this is happening, the program halts until that is complete, hence 
the "WebSocket ping".

It's possible this method can be improved, but hasn't been a high priority for the reasons mentioned above, harmless and reliable.

Obviously with everything going on in the world at the moment, it hasn't been a major concern of mine, but I do agree that it should be looked into. Also with my limited knowledge, this is the best I was able to produce when I started creating the program. And of course, if it is possible, it must also be just as reliable.

And if all that's not enough, OBS WebSockets 5.0.0 will change the way I approach things yet again. I'm currently waiting on a version of obs-websocket-js to test this with... been waiting a while, but everyone can only do so much, so I must be patient.

**Comment by bazz1tv** [comment]
I totally understand :)

**Comment by thenetimp** [comment]
Does VB not have a way to create the connection to the websocket itself that it's reliant on node to do it for it?  I came here for the same reason, not so much because of the spam, as I was able to disable the notification, but because it's actually causing OBS to crash.

Typically with a websocket, you open the connection, and keep it open reading in messages from the connection, and sending messages when you need too.  So it connecting constantly seemed really odd to me.
